src/ArrayToXml.php

```diff
@@ -212,4 +212,34 @@ protected function addAttributes($element, $data)
             $element->setAttribute($attrKey, $attrVal);
         }
     }
+
+    /**
+     * Create the root element.
+     *
+     * @param  string|array $rootElement
+     * @return DOMElement
+     */
+    protected function createRootElement($rootElement)
+    {
+        $rootElementName = is_array($rootElement)
+            ? $rootElement['rootElementName'] ?? ''
+            : $rootElement;
+
+        $element = $this->document->createElement($rootElementName == '' ? 'root' : $rootElementName);
+
+        if (is_array($rootElement)) {
+            foreach ($rootElement as $key => $value) {
+                switch ($key) {
+                    case '_attributes':
```

Comment:
Could you rewrite this with `if`s and  `continue`.

My guess is that it would feel a bit lighter.